rust/README.md

```diff
@@ -0,0 +1,96 @@
+# lib/rust
+
+Earthly's official collection of rust [UDCs](https://docs.earthly.dev/docs/guides/udc).
+
+## +CARGO
+
+This UDC runs the cargo command `cargo $args` caching the contents of `$CARGO_HOME/registry`, `$CARGO_HOME/git` and `target` for future builds of the same calling target. 
+
+### Arguments
+
+#### `args`
+Cargo subcommand and its arguments. Required
+
+#### `keep_fingerprints (false)`
+Do no remove source packages fingerprints. Use only when source packages have been `COPY`ed with `--keep-ts` option.
+
+Cargo caches compilations of packages in `target` folder based on their last modification timestamps. 
+
+By default, this UDC removes the fingerprints of the packages found in the source code, to force their recompilation and work even when the Earthly `COPY` commands used overwrote the timestamps.
+
+#### `output`
+Regex to match the files within the target folder to be copied from the cache to the caller filesystem (image layers). 
+
+Use this argument when you want to `SAVE AS ARTIFACT` from the target folder (mounted cache), always trying to minimize the total size of the copied fileset. 
+
+For example `--output="release/[^\./]+"` would keep all the files in /target/release that don't have any extension
+
+### Examples:
+
+Suppose the following project:
+```
+.
+├── Cargo.lock
+├── Cargo.toml
+├── deny.toml
+├── Earthfile
+├── package1
+│   ├── Cargo.toml
+│   └── src
+│       └── ...
+└── package2
+    ├── Cargo.toml
+    └── src
+        └── ...
+```
+
+The Earthfile would like:
+
+```earthfile
+VERSION 0.7
+ARG --global debian = bookworm
+
+# Importing via commit hash pinning because git tags can be changed
+IMPORT github.com/earthly/lib/rust:4cfebf74b5805ad943d325a94601e808afbf6e6f AS rust
```

Comment:
I know that tags can be changed, but changing them for released software is something you don't normally do. It might be good to start maintaining some official semver versions + tags for these subpackages. It would help users to keep track a bit better.

Perhaps the tags can be specific to a sub-package. e.g. `rust-0.1.0`.

Reply:
Great idea!

Reply:
@vladaionescu  why would we need semvers for specific packages and not for the whole repo?

Reply:
Mainly because the compatibility guarantees might be better if they're isolated. Like if we break compatibility for `rust`, why have the `go` users worry about it?

Reply:
yeah, but users can always look at the release notes, no? I just think it makes it harder (or at least eventually harder) to maintain

Reply:
Basically we would be breaking the semantics we are aiming to achieve. Is it truth that in a soft-way, but still we would be sending a confusing message: 
- Let's say I introduce a breaking change in `rust`, and increase the major version, then `go` users would perceive such version change and lose their stream of automatic updates
- As an API user it would be confusing and disappointing for me performing version updates that actually don't include any improvements.

Reply:
@idelvall , @vladaionescu 
I suggest having a 5 minute discussion on this in our next core/work chats (or ad-hoc if we don't want to wait until then) before we commit to this.

Reply:
Sounds good - booked!